stage0/src/image_header.rs

```diff
@@ -132,25 +42,41 @@ impl ImageHeader {
         // Next make sure the marked image length is programmed
         let valid = lpc55_romapi::validate_programmed(
             img_start,
-            (self.image_length + 0x1ff) & !(0x1ff),
+            (self.0.total_image_len + 0x1ff) & !(0x1ff),
         );
 
         if !valid {
             return false;
         }
 
+        if self.0.magic != abi::HEADER_MAGIC {
+            return false;
+        }
+
         return true;
     }
 
-    fn get_cert_table(&self) -> *const CertHeader {
-        self.get_table_start() as *const CertHeader
+    pub fn get_vectors(&self) -> u32 {
+        self.0.vector
     }
 
     pub fn get_pc(&self) -> u32 {
-        self.pc
+        let vec = self.get_vectors();
```

Comment:
`get_pc` needs to be `unsafe` now, because it can only be used safely from a context where you know the ImageHeader is intended to be valid. (As opposed to declaring a nonsense one on the stack and then using this to dereference nonsense pointers from apparently safe code.)

The `unsafe` should propagate up to somewhere that we're confident the header is as valid as we can achieve.

In practice it looks like `get_pc` is only called from `unsafe` code that is already operating under this requirement, so, should be an easy change.